### DISCUSSION TITLE
光机write_pattern_table函数中参数配置的一些疑问

### DISCUSSION THREAD

**Discussion by HaiMianBBao** [Open3DV/Xema#8] [Q&A] [answered]
# 疑问位于lightcrafter3010.cpp文件中的如下代码
```
int LightCrafter3010::write_pattern_table(unsigned char* pattern_index, unsigned char* pattern_nums, int len,float camera_exposure)
{
    // ......
    int pre_illumination_dark_time = 8500;
    int pose_illumination_dark_time = 1500;
    int exposure_rate = camera_exposure_ / 2000;
    pre_illumination_dark_time = 1000 + 170 * (exposure_rate);
    int illumination_time = camera_exposure - 1000;
    // ......
}
```
上述方法是为光机配置投影序列图，其中有几个参数配置不是很明白，分别是pre_illumination_dark_time，pose_illumination_dark_time ，exposure_rate ，pre_illumination_dark_time ，illumination_time 。

**Comment by zhangguanjin** [comment, marked as answer]
pre_illumination_dark_time，illumination_time ，pose_illumination_dark_time 是光机配置条纹投影序列需要设置的参数。
![1684120153546](https://github.com/Open3DV/Xema/assets/20849805/df80f52b-9483-4372-b956-7706ef358fc7)
相机采图与光机同步的方式是光机触发相机采集，当光机投影帧率大于相机采图帧率时，需要加大pre_illumination_dark_time值来降低光机的投影帧率，使相机与光机能正常同步。exposure_rate是一个测试出来的经验值，用于动态调节pre_illumination_dark_time 值。
